src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,324 @@
+        "Accepts regular expression. "
+        "If none are specified, all tensors are interpolated.")
+
+    parser.add_argument("--target-extents",
+                        type=int,
+                        nargs='+',
+                        required=True,
+                        help="The extents of the target grid. "
+                        "Can be different for each dimension e.g. '3 5 4'")
+
+    parser.add_argument("--target-basis",
+                        type=str,
+                        nargs='+',
+                        choices=basis_dict.keys(),
+                        default="Legendre",
+                        help="The basis of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument("--target-quadrature",
+                        type=str,
+                        nargs='+',
+                        choices=quadrature_dict.keys(),
+                        default="GaussLobatto",
+                        help="The quadrature of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument(
+        "--start-time",
+        type=float,
+        default=-np.inf,
+        help="Disregard all observations with value before this point")
+
+    parser.add_argument(
+        "--stop-time",
+        type=float,
+        default=np.inf,
+        help="Disregard all observations with value after this point")
+
+    parser.add_argument(
+        "--stride",
+        type=float,
+        default=1,
+        help="Stride through observations with this step size.")
+
+    parser.add_argument(
+        "-j",
+        "--jobs",
+        type=int,
+        default=None,
+        help="The maximum number of processes to be started. "
+        "A process is spawned for each source file up to this number.")
+
+    parser.add_argument('-v',
+                        '--verbose',
+                        action='count',
+                        default=0,
+                        help="Verbosity (-v, -vv, ...)")
+
+    return parser.parse_args(sys_args)
+
+
+if __name__ == "__main__":
+
+    parsed_args = parse_args(sys.argv[1:])
+
+    logging.basicConfig(level=logging.WARNING - parsed_args.verbose * 10)
+    source_files = glob.glob(parsed_args.source_file_prefix + "[0-9]*.h5")
+    file_numbers = [
+        re.match(parsed_args.source_file_prefix + "([0-9]*).h5",
+                 source_file).group(1) for source_file in source_files
+    ]
+    target_files = [
+        "{}{}.h5".format(parsed_args.target_file_prefix, number)
+        for number in file_numbers
+    ]
+    source_volumes = [parsed_args.source_subfile_name] * len(source_files)
+    target_volumes = [parsed_args.target_subfile_name] * len(target_files)
+
+    # construct target mesh
+    target_basis = basis_from_string(parsed_args.target_basis)
+    target_quadrature = quadrature_from_string(parsed_args.target_quadrature)
+    dim = len(parsed_args.target_extents)
```

Comment:
Maybe grab the dimension out of one of the source files, so it's ok to just specify one value for extent/basis/quadrature on the command line and it gets repeated for all dimensions?

Reply:
its possible but you lose the power of interpolating to different extents/bases/quadratures along different dimensions. I think this it is in line with our yaml input files, at least for the extents. Basis and quadrature is compile time so its hard to say.

Reply:
Oh I meant: Require the user specifies either a single value for all dims, or dim values (`assert len(extents) == 1 or len(extents) == dim`)

Reply:
Im pretty sure I had that in originally and we decided to remove it because it was one of the things that made the script too complex.